### DISCUSSION TITLE
Saving Repeater field entries gives "Call to undefined method Illuminate\Database\Eloquent\Relations\HasOne::associate()"

### DISCUSSION THREAD

**Discussion by theo-1** [filamentphp/filament#11372] [Help]
### Package

Form builder

### Package Version

v3.0.0

### How can we help you?

A record in the main table gets created but not the repeater field records. It gives the error:

`Call to undefined method Illuminate\Database\Eloquent\Relations\HasOne::associate()`

My InvoiceResource has within it:
```
...
Forms\Components\Section::make()
                    ->schema([
                        Forms\Components\Repeater::make('LineItems')
                            ->relationship('lineItems')  
...
```

My LineItemRelationsManager has within it:

`protected static string $relationship = 'lineItems';`

My Invoice model has within it:

`protected $primaryKey = 'InvoiceID';`

and 

```
public function lineItems(): HasMany {
        return $this->hasMany(Invoicelineitems::class,'InvoiceID', 'InvoiceID'); 
 }
```

Please how can I fix this?

**Comment by theo-1** [comment]
In select fields within the form I was using relationship() rather than options() with pluck. I switched them all from relationship() to options() and the error went away. But the documentation recommends using relationship in Panel Builder > Getting Started section. So what could be wrong with the usage here?

**Comment by micraux** [comment]
Did you create the inverse relationship in your `Invoicelineitems` model?
```php
public function invoice(): BelongsTo {
        return $this->BelongsTo(Invoice::class,'InvoiceID', 'InvoiceID'); 
 }
```
